pkg/pkg.mk

```diff
@@ -88,7 +93,14 @@ prepare: $(PKG_PREPARED)
 
 # Allow packages to add a custom step to be `prepared`.
 # It should be a dependency of `$(PKG_PREPARED)` and depend on `$(PKG_PATCHED)`
-$(PKG_PREPARED): $(PKG_PATCHED)
+# as well as `$(PKG_CUSTOM_POST_PATCH)`
+#
+# An example might be if you need to generate code after checkout, to be
+# compiled into the target application.  This needs to run before the
+# build step in order for header files to be available on the include path.
+# You can define a "PHONY" PKG_CUSTOM_POST_PATCH target in your package makefile
+# and execute whatever operation you need within it.
```

Comment:
I'm a bit worried that this is creating a foot-gun.
If you modify the code *after* checkout, it will be checked out again on the next build, so you are just wasting cycles - a normal patch would have been the better solution.

If you generate build for the target, you'd better be using `Makefile.gensrc` like `pkg/nanopb` otherwise your build will be very brittle.

It's a sufficiently hidden foot gun, but I'm still not quite convinced that this can be used for anything that won't blow up in your face eventually.

Reply:
@benpicco  thank you for the pointer to `Makefile.gensrc`.  I was unaware that existed.  After fiddling with that, the approach I needed (code generation prior to build) is achievable.  

I can abandon this PR.